addons/wowl/static/src/components/dropdown/dropdown.ts

```diff
@@ -0,0 +1,159 @@
+import { Component, useState } from "@odoo/owl";
+import { OwlEvent } from "@odoo/owl/dist/types/core/owl_event";
+import { DropdownItemClickedPayload, ParentClosingMode } from "./dropdown_item";
+
+export class Dropdown extends Component {
+  static template = "wowl.Dropdown";
+  static props = {
+    open: {
+      type: Boolean,
+      optional: true,
+    },
+    hover: {
+      type: Boolean,
+      optional: true,
+    },
+    hoverOpenDelay: {
+      type: Number,
+      optional: true,
+    },
+    hoverCloseDelay: {
+      type: Number,
+      optional: true,
+    },
+    closeOnOutsideClick: {
+      type: Boolean,
+      optional: true,
+    },
+    dropdownClass: {
+      type: String,
+      optional: true,
+    },
+    togglerClass: {
+      type: String,
+      optional: true,
+    },
+  };
+  static defaultProps = {
+    open: false,
+    hover: false,
+    hoverOpenDelay: 100,
+    hoverCloseDelay: 500,
+    closeOnOutsideClick: true,
+    menuClass: "o_dropdown_menu",
+    togglerClass: "o_dropdown_toggler",
```

Comment:
not sure if those classnames should be the default, or if they should simply be hardcoded in the template (with the props having empty default to allow specifying additional classnames)